pylint_venv.py

```diff
@@ -1,63 +1,116 @@
+    cwd = os.getcwd()
+    exec_path = "Scripts" if IS_WIN else "bin"
+    if os.path.isfile(os.path.join(cwd, ".venv", exec_path, "activate")):
+        return os.path.join(cwd, ".venv")
+
+    return None
+
+
+def activate_venv(venv):
+    """Check for an active venv and add its paths to Pylint.
 
     Activate the virtual environment from:
-    - The `venv` param, if one is given.
-    - `VIRTUAL_ENV` environmental variable if set.
-    - a `.venv` folder in the current working directory
-    """
-    if venv is None:
-        venv = os.environ.get("VIRTUAL_ENV", None)
 
-    if venv is None:
-        cwd = os.getcwd()
-        exec_path = 'Scripts' if sys.platform == "win32" else 'bin'
-        if os.path.isfile(os.path.join(cwd, ".venv", exec_path, "activate")):
-            venv = os.path.join(cwd, ".venv")
-
-    if venv is not None:
-        os.environ["VIRTUAL_ENV"] = venv
-        os.environ["PATH"] = (
-            os.path.join(venv, "bin") + os.pathsep + os.environ.get("PATH", "")
-        )
-        base = venv
-        if sys.platform == "win32":
-            site_packages = os.path.join(base, "Lib", "site-packages")
+    - The *venv* param if one is given.
+    - ``VIRTUAL_ENV`` environmental variable if set.
+    - A ``.venv`` folder in the current working directory
+
+    """
+    if IS_PYPY:
+        site_packages = os.path.join(venv, "site-packages")
+    else:
+        if IS_WIN:
+            site_packages = os.path.join(venv, "Lib", "site-packages")
         else:
-            site_packages = os.path.join(
-                base, "lib", "python%s" % sys.version[:3], "site-packages"
-            )
+            pyver = f"{sys.version_info[0]}.{sys.version_info[1]}"
+            site_packages = os.path.join(venv, "lib", f"python{pyver}", "site-packages")
 
-        prev = set(sys.path)
-        active_site.addsitedir(site_packages)
-        sys.real_prefix = sys.prefix
-        sys.prefix = base
+    prev = set(sys.path)
+    site.addsitedir(site_packages)
+    sys.real_prefix = sys.prefix
+    sys.prefix = venv
 
-        # Move the added items to the front of the path, in place
-        new = list(sys.path)
-        sys.path[:] = [i for i in new if i not in prev] + [i for i in new if i in prev]
+    # Move the added items to the front of sys.path (in place!)
+    new = list(sys.path)
+    sys.path[:] = [i for i in new if i not in prev] + [i for i in new if i in prev]
 
 
 def inithook(venv=None):
-    """Activate a Virtual Environment if one is not active."""
-    if not is_venv():
-        activate_venv(site, venv)
+    """Add a virtualenv's paths to Pylint.
+
+    Use the environment *env* if set or auto-detect an active virtualenv.
+
+    """
+    if is_venv():
+        # pylint was invoked from within a venv.  Nothing to do.
+        return
+
+    if venv is None:
+        venv = detect_venv()
+
+    if venv is None:
+        return
+
+    print(f"\033[1;33mUsing env: {venv}\033[0m", file=sys.stderr)
```

Comment:
Will the escape sequences also work on Windows terminals?

Reply:
Seems they are not supported on Windows for the time being: https://en.wikipedia.org/wiki/ANSI_escape_code#DOS_and_Windows
So these should be removed from the string on Windows (or all platforms).

Reply:
Meh.  That’s why people use termcolor or click :D I didn’t want to add a dependency for that little piece of niceness.  I will removing coloring for all platforms.